Common/src/linear_solvers_structure.cpp

```diff
@@ -738,7 +828,17 @@ unsigned long CSysSolve<ScalarType>::Solve(CSysMatrix<ScalarType> & Jacobian, CS
     ScreenOutput = config->GetDeform_Output();
   }
 
-  CMatrixVectorProduct<ScalarType> *mat_vec = NULL;
+  /*--- For smoother-type solvers set the associated preconditioner type. ---*/
+
+  switch (KindSolver) {
+    case SMOOTHER_LINELET: KindPrecond = LINELET; break;
```

Comment:
Should we just have KindSolver SMOOTHER in the config now?

Reply:
I like this idea, yes!

Reply:
Done